### PR TITLE
Fix clippy 1.73 warnings

### PR DESCRIPTION
**Pull request by LucasSte**
I updated Rust in my computer and fixed the errors from clippy.

### CODE REVIEW

File: src/codegen/cfg.rs [hunk 1]
```diff
@@ -657,10 +657,10 @@ impl ControlFlowGraph {
                 dimensions, values, ..
             } => format!(
                 "constant {} [ {} ]",
-                dimensions
-                    .iter()
-                    .map(|d| format!("[{d}]"))
-                    .collect::<String>(),
+                dimensions.iter().fold(String::new(), |mut output, d| {
+                    let _ = write!(output, "[{d}]");
```
Comment (seanyoung) [inline src/codegen/cfg.rs:661]: is `let _  = ` necessary?

File: src/codegen/cfg.rs [hunk 2]
```diff
@@ -671,10 +671,10 @@ impl ControlFlowGraph {
                 dimensions, values, ..
             } => format!(
                 "{} [ {} ]",
-                dimensions
-                    .iter()
-                    .map(|d| format!("[{d}]"))
-                    .collect::<String>(),
+                dimensions.iter().fold(String::new(), |mut output, d| {
+                    let _ = write!(output, "[{d}]");
```
Comment (seanyoung) [inline src/codegen/cfg.rs:675]: is `let _  = ` necessary?

**Review by seanyoung — APPROVED** [review]

File: src/codegen/cfg.rs (hunk 1 above)
Comment (LucasSte) [reply src/codegen/cfg.rs:661]: It is:
```
warning: unused `std::result::Result` that must be used
   --> src/codegen/cfg.rs:661:21
    |
661 |                     write!(output, "[{d}]");
    |                     ^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: this `Result` may be an `Err` variant, which should be handled
    = note: `#[warn(unused_must_use)]` on by default
    = note: this warning originates in the macro `write` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Comment (seanyoung) [reply src/codegen/cfg.rs:661]: So your `let _ = ` swallows errors, that's not great. It should be use `.unwrap()` or `.expect()`.

Comment (LucasSte) [reply src/codegen/cfg.rs:661]: The `let _ = ` was clippy's suggestion. https://rust-lang.github.io/rust-clippy/master/index.html#/format_collect
I'll change it to unwrap.

**Pull request merged**